### PR TITLE
[ts2pant-general-calls] Patch 1: Add test fixtures and unit test stubs for general function calls

### PR DESCRIPTION
**Pull request by subsetpark**
## Patch 1: Add test fixtures and unit test stubs for general function calls

- Create expressions-calls.ts with 7+ exported functions covering general call patterns
- Each function uses declare'd callees with known signatures so type extraction produces correct declarations
- Run npm run test:update-snapshots to generate initial snapshot expectations (UNSUPPORTED for general calls)
- Add it.skip('translateCallExpr > should translate free function call as uninterpreted application') // PENDING: Patch 2
- Add it.skip('translateCallExpr > should translate method call with receiver as first argument') // PENDING: Patch 2
- Add it.skip('translateCallExpr > should translate zero-arity call as variable reference') // PENDING: Patch 2
- Add it.skip('translateCallExpr > should reject spread arguments with UNSUPPORTED') // PENDING: Patch 2

## Changes
- Create expressions-calls.ts with 7+ exported functions covering general call patterns
- Each function uses declare'd callees with known signatures so type extraction produces correct declarations
- Run npm run test:update-snapshots to generate initial snapshot expectations (UNSUPPORTED for general calls)
- Add it.skip('translateCallExpr > should translate free function call as uninterpreted application') // PENDING: Patch 2
- Add it.skip('translateCallExpr > should translate method call with receiver as first argument') // PENDING: Patch 2
- Add it.skip('translateCallExpr > should translate zero-arity call as variable reference') // PENDING: Patch 2
- Add it.skip('translateCallExpr > should reject spread arguments with UNSUPPORTED') // PENDING: Patch 2

## Gameplan Specification

```
module TS2PANT_GENERAL_CALLS.

> After milestone 2: general call translation + purity oracle.
> Ref: Kroening & Strichman, Decision Procedures, Ch. 4 (EUF).
> Ref: Cousot & Cousot, Abstract Interpretation, POPL 1977.
> Ref: Lucassen & Gifford, Polymorphic Effect Systems, POPL 1988.

CallExpression.
ConstBinding.
PurityTier.

builtin-allowlist => PurityTier.
effect-ts-aware => PurityTier.
conservative-default => PurityTier.

special-cased? c: CallExpression => Bool.
has-spread? c: CallExpression => Bool.
identifier-callee? c: CallExpression => Bool.
prop-access-callee? c: CallExpression => Bool.
translated? c: CallExpression => Bool.
rejected? c: CallExpression => Bool.
tier c: CallExpression => PurityTier.
pure? c: CallExpression => Bool.
initializer b: ConstBinding => CallExpression.
inlineable? b: ConstBinding => Bool.

---

> Every call expression has a definite translation outcome.
all c: CallExpression |
  translated? c or rejected? c.

> Special cases always translate.
all c: CallExpression, special-cased? c |
  translated? c.

> General calls with known callee shape and no spread translate.
all c: CallExpression, ~special-cased? c, ~has-spread? c |
  (identifier-callee? c or prop-access-callee? c)
  -> translated? c.

> Spread in non-special calls is rejected.
all c: CallExpression, has-spread? c, ~special-cased? c |
  rejected? c.

> Purity tiers: allowlist and effect-ts are pure.
all c: CallExpression, tier c = builtin-allowlist |
  pure? c.

all c: CallExpression, tier c = effect-ts-aware |
  pure? c.

> Conservative default is not pure.
all c: CallExpression, tier c = conservative-default |
  ~pure? c.

> Const bindings with pure call initializers are inlineable.
all b: ConstBinding, pure? (initializer b) |
  inlineable? b.

> Const bindings with impure initializers are not inlineable.
all b: ConstBinding, ~pure? (initializer b) |
  ~inlineable? b.

```

## Patch Specification

```
module TS2PANT_GENERAL_CALLS_PATCH_1.

> Test fixtures and stubs only. No behavioral invariants.

CallExpression.

---

```

## Files to Modify
- tools/ts2pant/tests/fixtures/constructs/expressions-calls.ts (create): Fixture file with exported functions exercising general call patterns: freeCall (2-arg), zeroArityCall (0-arg), nestedCalls, callInArithmetic, callWithPropArg, methodCall, callInComparison. Uses declare'd callees.
- tools/ts2pant/tests/translate-body.test.mts (modify): Add skipped unit tests for translateCallExpr edge cases

## Implementation Notes

- Added 8 fixture functions (one more than the spec's 7): included `spreadCall` to exercise the spread-rejection path, since Patch 2 needs a test target for the `has-spread? → rejected?` invariant.
- All `declare`d callees (`max`, `now`, `clamp`, `validate`) have explicit signatures so the snapshot pipeline extracts correct Pantagruel declarations even though the call bodies are currently UNSUPPORTED. This validates that type extraction works independently of body translation.
- `methodCall` uses `s.toUpperCase()` — a built-in `String` method with no `declare` needed. This covers the property-access callee shape distinctly from the identifier callee shape used by free functions.
- Snapshots confirm the existing catch-all at `translateCallExpr` line 938 (`return { unsupported: expr.getText() }`) fires for every non-special call. Patch 2 will replace these UNSUPPORTED baselines with EUF translations.
- The 4 skipped tests in `translate-body.test.mts` are intentionally empty bodies with comments — Patch 2 will fill in the `createSourceFileFromSource` + `translateBody` assertions following the existing test pattern in that file.

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test coverage for call-expression translation scenarios.
  * Added fixtures exporting functions exercising free, zero-arity, nested, arithmetic, comparison, property-arg, method, and spread-call patterns.
  * Added skipped unit tests asserting expected translation behaviors (including rejecting spread args).
  * Appended snapshot entries recording expected outputs (including UNSUPPORTED outcomes for unsupported call shapes).

### CODE REVIEW

File: tools/ts2pant/tests/fixtures/constructs/expressions-calls.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+/** call result used in arithmetic */
+export function callInArithmetic(a: number, b: number): number {
+  return max(a, b) + 1;
+}
+
+/** call with property access argument */
+export function callWithPropArg(a: Account): boolean {
+  return validate(a);
+}
```
[the remark this answers is not in the document]
Comment (subsetpark) [reply tools/ts2pant/tests/fixtures/constructs/expressions-calls.ts:37]: Good catch — `validate(a)` was just passing an identifier, duplicating coverage from `freeCall`. Fixed: changed the declare to `validate(balance: number): boolean` and the call to `validate(a.balance)` so it actually exercises a property-access expression as a call argument. Snapshot updated accordingly.
